mitmproxy/net/http/request.py

```diff
@@ -441,7 +441,7 @@ def _set_urlencoded_form(self, form_data):
         This will overwrite the existing content if there is one.
         """
         self.headers["content-type"] = "application/x-www-form-urlencoded"
-        self.content = mitmproxy.net.http.url.encode(form_data, self.content.decode()).encode()
+        self.content = mitmproxy.net.http.url.encode(form_data, self.get_text(strict=False)).encode()
```

Comment:
Why `.content = (...).encode()` and not `.text = `? `.encode()` assumes the body should be utf8, with doesn't have to be true.

Reply:
I am sorry I didn't get what you are asking

Reply:
Instead of 
```
self.content = mitmproxy.net.http.url.encode(form_data, self.get_text(strict=False)).encode()
```
we can do
```
self.text = mitmproxy.net.http.url.encode(form_data, self.get_text(strict=False))
```
While it's admittedly hard to think of a case where urlencoded content cannot be `.encode()`d, it's more consistent to just always use `.text` for string assigments.